packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -176,6 +176,39 @@ interface PatchableModule {
   staticGenerationAsyncStorage: StaticGenerationAsyncStorage
 }
 
+/*
+ * When the fetch throws an error, the error stack trace will be:
+ * `Error: fetch failed` + node-internal trace -> `Error: fetch failed` + userland trace
+ */
+function traceErroredFetcher(
+  fetchFn: typeof fetch,
+  tracingError: Error
+): typeof fetch {
+  // Remove the error message and the trace line inside `patchFetch()`
+  const modifiedTrace = tracingError.stack?.split('\n').slice(2).join('\n')
+  const tracedFetch: typeof fetch = function (...args) {
+    return fetchFn(...args).catch((err) => {
+      // If it's failed with internal fetch call, and there's only node:internal traces
+      if (
+        err instanceof Error &&
+        err.name === 'TypeError' &&
+        err.message === 'fetch failed'
+      ) {
+        const traces: string[] = err.stack?.split('\n') || []
+        // trace without the error message
+        const originStackTrace = traces.slice(1)
+        if (originStackTrace.every((line) => line.includes('node:internal'))) {
```

Comment:
Why bother at all with the internal stack? Just do this for every error and every error message? What matters is it came from fetch.